app/src/main/java/com/duckduckgo/app/httpsupgrade/api/HttpsUpgradeDataDownloader.kt

```diff
@@ -107,4 +113,22 @@ class HttpsUpgradeDataDownloader @Inject constructor(
 
     }
 
+    fun reportUpgradeStatistics(): Completable {
+        return fromAction {
+
+            if (statisticsDataStore.httpsUpgradesTotal == 0) {
+                return@fromAction
+            }
+            val params = mapOf(
+                TOTAL_COUNT to statisticsDataStore.httpsUpgradesTotal.toString(),
+                FAILURE_COUNT to statisticsDataStore.httpsUpgradesFailures.toString()
+            )
+
+            pixel.fireCompletable(Pixel.PixelName.HTTPS_UPGRADE_SITE_SUMMARY, params).blockingGet()
```

Comment:
the `blockingGet()` is a bit of a smell here. We shouldn't need to subscribe inside a stream like this.

Perhaps instead, we should wrap this in `Completable.defer` instead of `Completable.fromAction` as we're currently doing. That would then become 👇 

        fun reportUpgradeStatistics(): Completable {
        return Completable.defer {
            if (statisticsDataStore.httpsUpgradesTotal == 0) {
                return@defer complete()
            }
            val params = mapOf(
                TOTAL_COUNT to statisticsDataStore.httpsUpgradesTotal.toString(),
                FAILURE_COUNT to statisticsDataStore.httpsUpgradesFailures.toString()
            )

            return@defer pixel.fireCompletable(Pixel.PixelName.HTTPS_UPGRADE_SITE_SUMMARY, params).andThen {
                Timber.v("Sent https statistics")
                statisticsDataStore.httpsUpgradesTotal = 0
                statisticsDataStore.httpsUpgradesFailures = 0
            }
        }
    }

Note, the use of `andThen` to hook in the action to be done after it was successfully upload. 

Test it out, of course, to make sure I haven't screwed anything up! 😄 But I think that's _cleaner_ in Rx world.

Reply:
Great idea!